compiler/src/dotty/tools/dotc/util/SourceFile.scala

```diff
@@ -211,8 +211,8 @@ class SourceFile(val file: AbstractFile, computeContent: => Array[Char]) extends
   def startColumnPadding(offset: Int): String = {
     var idx = startOfLine(offset)
     val pad = new StringBuilder
-    while (idx != offset) {
-      pad.append(if (idx < length && content()(idx) == '\t') '\t' else ' ')
+    while (idx != offset && idx < length && idx < content().length) {
```

Comment:
This handles the case when `content().length != length`

Reply:
@nicolasstucki Could you have a look at this change?

Reply:
I don't see how `content().length != length` is possible. `calculateLineIndicesFromContents()` has:
```scala
val cs = content()
...
buf += cs.length
```
and `length` calls `lineIndicesCache.last`.

@Xavientois do you have a concrete test case for which `content().length != length`?

Reply:
It seems content().length != length happens when length is computed from Tasty using setLineIndicesFromLineSizes instead of using calculateLineIndicesFromContents.